pkg/client/client.go

```diff
@@ -887,8 +887,10 @@ func (c *ConmonClient) ExecSyncContainer(ctx context.Context, cfg *ExecSyncConfi
 
 	execContainerResult := &ExecContainerResult{
 		ExitCode: resp.ExitCode(),
-		Stdout:   stdout,
-		Stderr:   stderr,
+		// Copy content of both slices as the `capnp.ReleaseFunc` will
+		// zero them when the deferred call to `free()` takes place.
+		Stdout:   append(stdout[:0:0], stdout...),
+		Stderr:   append(stderr[:0:0], stderr...),
```

Comment:
**To the reviewer:**

Since the changes from the following Pull Request:

- https://github.com/capnproto/go-capnp/pull/489

Following these changes referenced above, the `ReleaseFunc` would clean (or reset, if you wish) the data the promise returns. This is a problem as slices are passed using a reference, rather than copied like the primitive types. As such, the slice which holds the data will be reset to its default state behind the scenes once the `free()` callback from the affected code is invoked.

Thus, we have two options:

1. Copy the content of the stderr and stdout slices
1. Change the API such that the function would return the `ReleaseFunc` callback too

The #1 is simple and keeps the implementation details still hidden, but would waste a little bit of memory in the process, whereas the #2 would require calls to not forget to call the `ReleaseFunc` once the results are no longer needed.

Reply:
Let's go with option 1 for now.